### ISSUE TITLE
About the recreation of the paper experiment

### ISSUE THREAD

**Issue by NguyenTuan9347** [Graph-COM/GAD-NR#4]
Hello when I read into your code specifically for all benchmark, data used for both evaluating and train is the same dataset and in the original paper there are no mention of the split ratio or where i can find the test and train dataset to get the result in your paper. Can you give me some guild on how I can get / recreate your test and train dataset. With warm regard.

**Comment by AmitRoy7781** [comment]
For this work, the combined reconstruction loss is used to calculate the roc-auc between normal and anomaly nodes without spliting into training, validation and test set. Thanks.

**Comment by apu20nam** [comment]
@AmitRoy7781 @lipan00123 Please can you explain why you didn't split  train and test. And if you are calculation roc-auc score between normal and anomaly nodes by giving whole nodes into the train then how can we say that this is best result .

**Comment by AmitRoy7781** [comment]
For this paper, we followed the same setting as the benchmarking outlier node detection ([BOND](https://arxiv.org/pdf/2206.10071)) paper to compare with baseline GAD approaches, where the data is not split into training, validation, and test set and the best performance is reported over multiple runs. 

For a practical scenario, the training of anomaly detection models is done on the available training data and since the model is unsupervised it can be applied to unseen graphs if it attains a certain level of performance on the training data.

**Comment by apu20nam** [comment]
@AmitRoy7781 @lipan00123 Please can you explain about the running time you mentioned in the paper is training time or inference time . if it is inference time how you applied because there has no mention about test data. Please explain it .

**Comment by AmitRoy7781** [comment]
@apu20nam  We reported per epoch training time.

**Comment by apu20nam** [comment]
@AmitRoy7781 and @lipan00123 Could you explain why the best_auc is used to determine the AUC score? Shouldn't the AUC be calculated after the model completes all iterations? This approach of using best_auc rather than the final AUC score seems questionable. In practice, for larger networks, the best_auc sometimes occurs at an early iteration when the loss is still very high, while the final AUC score might be lower even when the loss is minimal or at the end of training. Please clarify the intuition behind this method.

![Image](https://github.com/user-attachments/assets/30c098ea-6f46-4587-84b3-9e1014a3417f)

![Image](https://github.com/user-attachments/assets/78412af4-d275-4919-a005-318f1f5ac045)

**Comment by AmitRoy7781** [comment]
To compare with baseline approaches, we reported the best AUC score during the training. Note that, for the anomaly detection problem we don't have any test data in this setting. Therefore, we choose the best auc score while training the GAD-NR model to compare with baseline approaches.
